can/src/stream.rs

```diff
@@ -416,21 +416,29 @@ mod imp {
         frame: &mut RawFrame<N>,
         flags: libc::c_int,
         src_addr: T,
-    ) -> io::Result<usize> {
+    ) -> io::Result<()> {
         let (addr, addrlen) = src_addr.to_recv_from_arguments();
-        let ret = unsafe {
+        let nbytes = unsafe {
             libc::recvfrom(
                 fd.as_raw_fd(),
-                (frame as *mut RawFrame<N>) as *mut libc::c_void,
-                std::mem::size_of::<RawFrame<N>>(),
+                std::ptr::from_mut(frame).cast::<libc::c_void>(),
+                std::mem::size_of::<RawFrame<{ N }>>(),
                 flags,
                 addr,
                 addrlen,
             )
         };
-        if ret < 0 {
+        if nbytes < 0 {
             return Err(io::Error::last_os_error());
+        } else if nbytes as usize != std::mem::size_of::<RawFrame<{ N }>>() {
```

Comment:
This does not hold up, as when `N = 64` the acceptable values of `nbytes` are either `size_of::<RawFrame<N=8>>` OR `size_of::<RawFrame<N=64>>`